packages/muelu/src/Utils/MueLu_Utilities_decl.hpp

```diff
@@ -1152,23 +1157,57 @@ EpetraMultiVector_To_XpetraMultiVector(const Teuchos::RCP<Epetra_MultiVector>& V
   TODO move this function to an Xpetra utility file
   */
 template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >
-TpetraCrs_To_XpetraMatrix(const Teuchos::RCP<Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& Atpetra);
+RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>>
+TpetraCrs_To_XpetraMatrix(const Teuchos::RCP<Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>>& Atpetra);
 
 /*! \fn TpetraMultiVector_To_XpetraMultiVector
   @brief Helper function to convert a Tpetra::MultiVector to an Xpetra::MultiVector
   TODO move this function to an Xpetra utility file
   */
 template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-RCP<Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> >
-TpetraMultiVector_To_XpetraMultiVector(const Teuchos::RCP<Tpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& Vtpetra);
+RCP<Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>>
+TpetraMultiVector_To_XpetraMultiVector(const Teuchos::RCP<Tpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>>& Vtpetra);
 
 // Generates a communicator whose only members are other ranks of the baseComm on my node
-Teuchos::RCP<const Teuchos::Comm<int> > GenerateNodeComm(RCP<const Teuchos::Comm<int> >& baseComm, int& NodeId, const int reductionFactor);
+Teuchos::RCP<const Teuchos::Comm<int>> GenerateNodeComm(RCP<const Teuchos::Comm<int>>& baseComm, int& NodeId, const int reductionFactor);
 
 // Lower case string
 std::string lowerCase(const std::string& s);
 
+template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+Teuchos::RCP<const Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node>> importOffRankDroppingInfo(
+    Teuchos::RCP<const Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node>>& localDropMap,
+    Teuchos::RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>>& Ain) {
+  using SC = Scalar;
+  using LO = LocalOrdinal;
+  using GO = GlobalOrdinal;
+  using NO = Node;
+
+  Teuchos::RCP<const Teuchos::Comm<int>> comm = localDropMap->getComm();
+
+  Teuchos::RCP<Xpetra::Vector<SC, LO, GO, NO>> toggleVec = Xpetra::VectorFactory<SC, LO, GO, NO>::Build(localDropMap);
+  toggleVec->putScalar(1);
+
+  Teuchos::RCP<Xpetra::Vector<SC, LO, GO, NO>> finalVec = Xpetra::VectorFactory<SC, LO, GO, NO>::Build(Ain->getColMap(), true);
```

Comment:
Doesn't this mean that all entries will be nonzero and consequently be dropped? Is this missing a step where some entries of `toggleVec` are set to zero?

Reply:
In the map-pair dropping scheme in the SegregatedAFactory entries of the matrix need to be dropped if a row/column pair is identified in the the two dropping maps. 

The toggleVec is defined on the local map. All entries contained in this map are potential dropping candidates, hence all entries in toggleVec are set to 1.

The finalVec is defined on the column map of the matrix. This map is “larger” than the local map. After the import, the finalVec contains some “1” entries (these entries originate from the toggleVec, but have been properly communicated through by the import operation) and lots of “0” entries.

It is important to note that the map of toggleVec is a subset of the map of finalVec.

Then, the sparsity pattern of finalVec is used to define the entries of the finalDropMap, which has been extended and now contains some off-Rank entries.

I demonstrate this behavior in the unit test SegregatedAFactory(importOffRankDroppingInfo). There you can see that on proc 4 the expectedMap only contains entries 7, 8 as there is no local entry in the column map of the matrix. All the others however have entries 1, 7 and 8 and have been communicated properly. 

I hope this make things more clear.

Reply:
Ah shoot. I totally forgot that the map is a submap of the rowmap. Yes, that does work.